Jenkinsfile

```diff
@@ -57,111 +56,202 @@ spec:
+                        }
+                        post {
+                            success{
+                                println "Testing reponse app tag was successful"
+                            }
+                            failure{
+                                println "Error testing response app tag"
+                            }
+                        }
+                    }
+                    stage('Test read and write') {
+                        steps {
+                            container('docker') {
+                                script {
+                                    HTTP_RESPONSE_CODE_1 = sh (script: 'docker run -i --net=curltest tutum/curl \
+                                        curl -o /dev/null -I -s -w "%{http_code}" http://dropw-test:8080/hello-world', returnStdout: true).trim()
+                                    HTTP_RESPONSE_CODE_2 = sh (script: 'docker run -i --net=curltest tutum/curl \
+                                        curl -H "Content-Type: application/json" -o /dev/null -s -w "%{http_code}" -X POST -d \'{"fullName":"Test Person","jobTitle":"Test Title"}\' http://dropw-test:8080/people', returnStdout: true).trim()
+                                    HTTP_RESPONSE_CODE_3 = sh (script: 'docker run -i --net=curltest tutum/curl \
+                                        curl -o /dev/null -I -s -w "%{http_code}" http://dropw-test:8080/people/1', returnStdout: true).trim()
+                                    if (!"${HTTP_RESPONSE_CODE_1}" == 200 || !"${HTTP_RESPONSE_CODE_2}" == 200 || !"${HTTP_RESPONSE_CODE_3}" == 200) {
+                                        println "Raising failure status"
+                                        throw new Exception("Testing read and write failure!")
+                                    }
+                                }
+                            }
+                        }
+                        post {
+                            success{
+                                println "Testing successful."
+                            }
+                            failure{
+                                println "Testing is not completed:"
+                                println "HTTP response for GET hello-world page is - ${HTTP_RESPONSE_CODE_1}"
+                                println "HTTP response for POST test person is - ${HTTP_RESPONSE_CODE_2}"
+                                println "HTTP response for GET test person - ${HTTP_RESPONSE_CODE_3}"
+                            }
+                        }
                     }
                 }
             }
         }
     }
 
-    stage('Deploy to k8s') {
+    stage('Docker tag image and push to repository.') {
+        when { not { changeRequest() } }
+        steps {
+            container('docker') {
+                script {
+                    IMAGE_NAME = ("${GIT_BRANCH}"=='master') ? "${DOCKERHUB_REPO}/${HELM_RELEASE}" : "${DOCKERHUB_REPO}/${HELM_RELEASE}-${GIT_BRANCH}"
```

Comment:
This isn't correct
docker image name shouldn't change no matter what branch we're building

Reply:
I've thought you implied this as one of the first requirment. 
OK, I'll move it into tag.

Reply:
Done